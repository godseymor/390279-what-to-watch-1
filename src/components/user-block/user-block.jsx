import {bool, func, number, string, shape} from "prop-types";
import React, {PureComponent} from "react";
import {compose} from "redux";
import {withRouter} from "react-router";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

class UserBlock extends PureComponent {
  constructor(props) {
    super(props);
    this._handelAvatarClick = this._handelAvatarClick.bind(this);
  }

  render() {
    const {authorized, user} = this.props;

    if (!authorized) {
      return (
        <div className="user-block">
          <Link to="/login" className="user-block__link">
            Sign in
          </Link>
        </div>
      );
    } else {
      return (
        <div className="user-block">
          <div className="user-block__avatar" onClick={this._handelAvatarClick}>
            <img
              src={`https://es31-server.appspot.com/${user.userAvatar}`}
              alt={user.userName}
              width="63"
              height="63"
            />
          </div>
        </div>
      );
    }
  }

  _handelAvatarClick() {
    const {history} = this.props;
    history.push(`/mylist`);
  }
}

UserBlock.propTypes = {
  authorized: bool.isRequired,
  user: shape({
    userId: number,
    userEmail: string,
    userName: string,
    userAvatar: string
  }).isRequired,
  history: shape({
    push: func.isRequired
  }).isRequired
};

const mapStateToProps = (state) => ({
  user: state.user.currentUser,
  authorized: state.user.authorized
});

export {UserBlock};

export default compose(
    connect(
        mapStateToProps,
        null
    ),
    withRouter
)(UserBlock);
