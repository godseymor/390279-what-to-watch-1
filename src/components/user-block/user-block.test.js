import React from "react";
import renderer from "react-test-renderer";
import {UserBlock} from "./user-block.jsx";

const mocks = {
  authorized: true,
  user: {
    userId: 1,
    userEmail: `userEmail`,
    userName: `userName`,
    userAvatar: `userAvatar`
  },
  history: {
    push: jest.fn()
  }
};

describe(`UserBlock:`, () => {
  it(`Ð¡orrectly rendered after reload`, () => {
    const tree = renderer.create(<UserBlock {...mocks} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
