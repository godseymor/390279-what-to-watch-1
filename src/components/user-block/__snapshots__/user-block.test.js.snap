// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock: Ð¡orrectly rendered after reload 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
    onClick={[Function]}
  >
    <img
      alt="userName"
      height="63"
      src="https://es31-server.appspot.com/userAvatar"
      width="63"
    />
  </div>
</div>
`;
